homeassistant/components/sabnzbd/sensor.py

```diff
@@ -39,33 +42,15 @@ async def async_added_to_hass(self):
             )
         )
 
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return f"{self._client_name} {self._name}"
-
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self._state
-
-    @property
-    def should_poll(self):
-        """Don't poll. Will be updated by dispatcher signal."""
-        return False
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement of this entity, if any."""
-        return self._unit_of_measurement
-
     def update_state(self, args):
         """Get the latest data and updates the states."""
-        self._state = self._sabnzbd_api.get_queue_field(self._field_name)
-
-        if self._type == "speed":
-            self._state = round(float(self._state) / 1024, 1)
-        elif "size" in self._type:
-            self._state = round(float(self._state), 2)
+        new_state = self._sabnzbd_api.get_queue_field(self._field_name)
+
+        if self.entity_description.key == "speed":
```

Comment:
This should now be this?

```suggestion
        if self.entity_description.key == "kbpersec":
```